Formula/terraform.rb

```diff
@@ -12,7 +12,7 @@ class Terraform < Formula
     sha256 "1e6a06b503baf84d70ba84a03609ee100e0fc2065853c8d211215b1fe27f9dd3" => :high_sierra
   end
 
-  depends_on "go" => :build
+  depends_on "go@1.13" => :build
```

Comment:
It looks like we'll be able to switch it back to the latest go for the next release: https://github.com/hashicorp/terraform/commit/43c1ec69e2739ebb036a7fe871d9ff0bde4acc59

Reply:
Not currently, the issue has been reported on master as well: https://github.com/hashicorp/terraform/issues/24287, but it will hopefully be fixed by the time 0.13 is released anyway, whether that be on Go or Terraform's part.